File: Java/problem1512.java
class Solution {
    public int numIdenticalPairs(int[] nums) {
        Map<Integer, Integer> count = new HashMap<>();
        for (int num : nums) {
            count.put(num, count.getOrDefault(num, 0) + 1);
        }
        int res = 0;
        for (int c : count.values()) {
            res += c * (c - 1) / 2;
        }
        return res;
    }
}

File: python/problem1512.py
class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        num2count = collections.defaultdict(int)
        for num in nums:
            num2count[num] += 1
        result = 0
        for num in num2count:
            result += (num2count[num] * (num2count[num] - 1)) // 2
        return result
